src/viam/logging.py

```diff
@@ -53,13 +85,34 @@ def addHandlers(logger: logging.Logger):
     # filter out logs at error level or above
     handler.setLevel(LOG_LEVEL)
     handler.addFilter(filter=lambda record: (record.levelno < ERROR))
-    logger.addHandler(handler)
 
     err_handler = logging.StreamHandler(stream=sys.stderr)
     err_handler.setFormatter(format)
     # filter out logs below error level
     err_handler.setLevel(max(ERROR, LOG_LEVEL))
-    logger.addHandler(err_handler)
+
+    if _MODULE_PARENT is not None and not default:
+        _startModuleLogging(logger)
+    else:
+        logger.addHandler(handler)
+        logger.addHandler(err_handler)
+
+
+def _startModuleLogging(logger: logging.Logger):
+    handler = ModuleHandler(logger)
+
+    format = ColorFormatter("%(asctime)s\t\t" + "%(levelname)s\t" + "%(name)s (%(filename)s:%(lineno)d)\t" + "%(message)s\t")
+    handler.setFormatter(format)
+    handler.setLevel(LOG_LEVEL)
+
+    logger.addHandler(handler)
+
+
+def setParent(parent):
```

Comment:
For this line and when we initialize (`_MODULE_PARENT = None`), parent is of type `RobotClient`. The problem with clarifying the type is that importing `RobotClient` introduces a circular import error, as `RobotClient` imports `viam.logging`.
Even if we put module handler in its own file, `logging.py` would import from `modulehandler`, which imports `RobotClient`, which imports `logging.py`, and etc.

Reply:
Gotcha... Not annotating seems fine; also just wanted to clarify what that type was for my own understanding. Thank you!

Reply:
You can actually do it by doing the following

```python3
from typing import TYPE_CHECKING
if TYPE_CHECKING:
    from .robot.client import RobotClient

def setParent(parent: "RobotClient"):
    ...
```